_posts/2023-02-01-opentelemetry.md

```diff
@@ -0,0 +1,214 @@
+
+While the OpenTracing project is strictly tight to the Jaeger implementation, by using the Jaeger client underneath, OpenTelemetry provides a different approach.
+The tracer uses an exporter in order to export traces in a specific format to the backend system.
+While a Jaeger based exporter is available, to allow current users to continue using the Jaeger protocol, the OTLP (OpenTeLemetry Protocol) one is used by default and the OpenTelemetry community encourage to move to it.
+You can also use Zipkin or developing your own exporter for a custom backend system.
+
+Using a specific exporter protocol means that the corresponding backend system has to support it.
+It has to provide an endpoint to which the traces are sent by using that specific protocol.
+For example, you can still use Jaeger backend with the OTLP exporter protocol, because newer Jaeger releases since 1.35 exposes an OTLP endpoint.
+You could also decide to use the usual Jaeger protocol endpoint by setting the corresponding Jaeger exporter in your system.
+
+From an application perspective, the way OpenTelemetry works is similar to OpenTracing (using Jaeger client).
+The tracer is part of your application and it provides the API for creating trace spans and using the configured exporter to send them.
+Focusing on Apache Kafka, the application could be a producer or consumer, or one of the components managed by the Strimzi operator, such as Kafka Connect, Kafka Mirror Maker(s) and the HTTP bridge.
+
+As it happens with OpenTracing, even the OpenTelemetry exporter doesn't send all traces but often just part of them by using sampling.
+The sampler mechanism can be configured through environment variables.
+
+## How to enable OpenTelemetry
+
+Before adding the support for OpenTelemetry, it was already possible to enable the distributed tracing on the `KafkaConnect`, `KafkaMirrorMaker`, `KafkaMirrorMaker2` and `KafkaBridge` custom resources by setting the `spec.tracing.type` property to `jaeger` to use OpenTracing.
+Since the Strimzi 0.32.0 release, you can set the corresponding `opentelemetry` value for using OpenTelemetry instead.
+
+```yaml
+# ...
+spec:
+  # ...
+  tracing:
+    type: opentelemetry
+  # ...  
+```
+
+With the above setting, the Strimzi operator initializes the OpenTelemetry tracer using the default OTLP exporter, for sending traces to the tracing backend.
+But you also need to specify environment variables to configure the exporter itself, by using the `spec.template` properties.
+Following, an example of the `KafkaBridge` configuration.
+
+```yaml
+spec:
+  #...
+  template:
+    bridgeContainer:
+      env:
+        - name: OTEL_SERVICE_NAME
+          value: my-otel-service
+        - name: OTEL_EXPORTER_OTLP_ENDPOINT
+          value: "http://jaeger-host:4317"
+  tracing:
+    type: opentelemetry
+```
+
+It is using the service name `my-otel-service` which will show up in the backend UI.
```

Comment:
Should you clarify that this should be different for different services / operands?